### DISCUSSION TITLE
How to decode instruction details?

### DISCUSSION THREAD

**Discussion by rottensunday** [otter-sec/anchor#3267] [Q&A]
Hello!
I have following use case:
- I have a transaction from Solana (result of `getTransaction`/`getBlock` RPC call)
- I want to decode instructions in this transaction. I expect some interface that allows me to pass instruction data, accounts, and get decoded instruction details. I'm able to do it for some SPL programs, like Token, with this: https://docs.rs/solana-transaction-status/1.18.23/solana_transaction_status/parse_instruction/index.html
- But I want more than SPL programs! I want to use an arbitrary IDL to be able to parse instructions. With this package, if I have IDL, I'm able to generate interface for given program (https://docs.rs/anchor-lang/latest/anchor_lang/macro.declare_program.html). Unfortunately, this interface has no methods to parse instructions. The closest thing I can find is `utils::Event::try_from_bytes`. But I don't care about events - I want instructions.

For example: I want to load Raydium IDL, and be able to parse Raydium swap instructions. Is there some solution in this package I'm missing? Or maybe there are better ways to achieve this?

I don't want to use Typescript.

Thank you for any help!

**Comment by BAXYCode** [comment]
commenting to pump this post as this is something I'm also looking for

**Comment by mikedotcook** [comment]
Literally facing the same problem rn 💯 Any ideas?

**Comment by rishenco** [comment]
I'm looking for this functionality too!

**Comment by kirillermakovichh** [comment]
any updates?
